docs/content/latest/architecture/docdb-sharding/tablet-splitting.md

```diff
@@ -195,21 +192,112 @@ Tablet UUID                       Range                                 Leader
+
+1. Create a three-node cluster.
+
+```sh
+./bin/yb-ctl --rf=3 create --num_shards_per_tserver=1 --ysql_num_shards_per_tserver=1 --master_flags '"tablet_split_size_threshold=30000000"' --tserver_flags '"memstore_size_mb=10"'
+```
+
+2. Create a table for workload.
+
+```sh
+./bin/ysqlsh -c "CREATE DATABASE ycsb;"
+./bin/ysqlsh -d ycsb -c "CREATE TABLE usertable (
+           YCSB_KEY TEXT,
+           FIELD0 TEXT, FIELD1 TEXT, FIELD2 TEXT, FIELD3 TEXT,
+           FIELD4 TEXT, FIELD5 TEXT, FIELD6 TEXT, FIELD7 TEXT,
+           FIELD8 TEXT, FIELD9 TEXT,
+           PRIMARY KEY (YCSB_KEY ASC));"
+```
+
+3. Create the properties file for YCSB at `~/code/YCSB/db-local.properties` and add the following content.
+
+```sh
+db.driver=org.postgresql.Driver
+db.url=jdbc:postgresql://127.0.0.1:5433/ycsb;jdbc:postgresql://127.0.0.2:5433/ycsb;jdbc:postgresql://127.0.0.3:5433/ycsb
+db.user=yugabyte
+db.passwd=
+core_workload_insertion_retry_limit=10
+```
+
+4. Start loading data for the workload.
+
+```sh
+~/code/YCSB/bin/ycsb load jdbc -s -P ~/code/YCSB/db-local.properties -P ~/code/YCSB/workloads/workloada -p recordcount=500000 -p operationcount=1000000 -p threadcount=4
+```
+
+5. Monitor the tablets splitting by going to the YugabyteDB Web UI at `http://localhost:7000/tablet-servers` and `http://127.0.0.1:9000/tablets`.
+
+6. After the data is loaded, save the per-tablet metrics into a file:
+
+The following assumes `curl` and `jq` are installed.
+
+```sh
+(curl -s http://127.0.0.1:9000/metrics && curl -s http://127.0.0.2:9000/metrics && curl -s http://127.0.0.3:9000/metrics) | jq -s '[flatten | .[] | select(.type == "tablet") as $tablet | .metrics[] | select(.name == "rocksdb_number_db_seek" and .value > 0) | { tablet_id: $tablet.id, rocksdb_number_db_seek: .value }] | sort_by (.tablet_id)' >after-load.json
+```
+
+7. Run the workload.
+
+```sh
+~/code/YCSB/bin/ycsb run jdbc -s -P ~/code/YCSB/db-local.properties -P ~/code/YCSB/workloads/workloada -p recordcount=500000 -p operationcount=1000000 -p threadcount=4
+```
+
+8. After the workload is completed, save the per-tablet metrics into another file.
+
+```sh
+(curl -s http://127.0.0.1:9000/metrics && curl -s http://127.0.0.2:9000/metrics && curl -s http://127.0.0.3:9000/metrics) | jq -s '[flatten | .[] | select(.type == "tablet") as $tablet | .metrics[] | select(.name == "rocksdb_number_db_seek" and .value > 0) | { tablet_id: $tablet.id, rocksdb_number_db_seek: .value }] | sort_by (.tablet_id)' >after-run.json
+```
+
+9. Get the list of tablets accessed during the run phase of the workload.
+
+```sh
+diff -C1 after-load.json after-run.json | grep tablet_id | sort | uniq
+```
+
+10. The list of tablets accessed can be compared with `http://127.0.0.1:9000/tablets` to make sure no pre-split tablets have been accessed.
```

Comment:
we also need a sub section of extra notes, eg: tablets will not split again until a full compaction, etc cc @ttyusupov

Reply:
added